flink-core/src/main/java/org/apache/flink/types/parser/FieldParser.java

```diff
@@ -174,6 +174,62 @@ protected void setErrorState(ParseErrorState error) {
 	public ParseErrorState getErrorState() {
 		return this.errorState;
 	}
+
+	/**
+	 * Returns the end position of a string with a numeric format (like XXXX-XX-XX). Sets the error state if the
+	 * string contains leading/trailing whitespaces or if the column is empty.
+	 *
+	 * @return the end position of the string or -1 if an error occurred
+	 */
+	public final int formattedStringEndPos(byte[] bytes, int startPos, int limit, byte[] delimiter) {
+		int len = startPos;
+
+		final int delimLimit = limit - delimiter.length + 1;
+
+		while (len < limit) {
+			if (len < delimLimit && delimiterNext(bytes, len, delimiter)) {
+				if (len == startPos) {
+					setErrorState(ParseErrorState.EMPTY_COLUMN);
+					return -1;
+				}
+				break;
+			}
+			len++;
+		}
+
+		if (len > startPos &&
+				(Character.isWhitespace(bytes[startPos]) || Character.isWhitespace(bytes[(len - 1)]))) {
+			setErrorState(ParseErrorState.NUMERIC_VALUE_ILLEGAL_CHARACTER);
+			return -1;
+		}
+
+		return len;
+	}
+
+	/**
+	 * Returns a string with a numeric format (like XXXX-XX-XX). Throws an exception if the
+	 * string contains leading/trailing whitespaces or if the column is empty.
+	 *
+	 * @return the parsed string
+	 */
+	public static final String formattedString(byte[] bytes, int startPos, int length, char delimiter) {
+		if (length <= 0) {
+			throw new NumberFormatException("Invalid input: Empty string");
```

Comment:
But this is not the `parseField` method. `parseField` could catch the exception and pass it on as a `NumberFormatException`.